test/gtest/ternary_tensor_ops.cpp

```diff
@@ -0,0 +1,452 @@
+                                                       {1, 1, 16, 8},
+                                                       {1, 1, 1, 8},
+                                                       {20, 16, 8},
+                                                       {20, 16, 1},
+                                                       {1, 16, 8},
+                                                       {1, 16, 1},
+                                                       {20, 1, 1},
+                                                       {16, 8},
+                                                       {16, 1},
+                                                       {1, 8},
+                                                       {8},
+                                                       {1}};
+
+static std::vector<std::vector<int64_t>> offsetsArr = {
+    {0, 0, 0}, {64, 32, 16}, {32, 16, 32}, {32, 16, 32}};
+
+static std::vector<std::vector<float>> alphabetaArr = {{1, 1, 0}, {-1, 1, 1}, {1.0, 0.5, 0.3}};
+
+static std::vector<std::vector<int>> stridesArr = {{8 * 16 * 20 * 16, 8 * 16 * 20, 8 * 16, 8, 1}};
+
+static std::vector<bool> packedArr = {true, false};
+
+static std::vector<miopenTensorOp_t> operationArr = {
+    miopenTensorOpAdd, miopenTensorOpMul, miopenTensorOpMin, miopenTensorOpMax};
+
+struct TestCase
+{
+    std::vector<int> tensorlens_ac;
+    std::vector<int> tensorlens_b;
+    std::vector<int64_t> offsets;
+    std::vector<int> stride_a;
+    std::vector<int> stride_b;
+    std::vector<int> stride_c;
+    std::vector<float> alphabeta;
+    bool packed;
+    miopenTensorOp_t operation;
+};
+
+template <typename T>
+struct TensorOpsCommon : public testing::TestWithParam<TestCase>
+{
+    void SetUp() override
+    {
+        prng::reset_seed();
+
+        CreateTensors();
+
+        tensor<T> tensorGPU = std::move(CalculateOnGPU());
+        tensor<T> tensorCPU = std::move(CalculateOnCPU());
+
+        CompareResults(tensorGPU, tensorCPU);
+    }
+
+private:
+    void CreateTensors()
+    {
+        const TestCase& testCase = GetParam();
+
+        tensorA = std::move(CreateTensor(
+            testCase.tensorlens_ac, testCase.stride_a, testCase.offsets[0], testCase.packed));
+        tensorB = std::move(CreateTensor(
+            testCase.tensorlens_b, testCase.stride_b, testCase.offsets[1], testCase.packed));
+        tensorC = std::move(CreateTensor(
+            testCase.tensorlens_ac, testCase.stride_c, testCase.offsets[2], testCase.packed));
+    }
+
+    tensor<T> CreateTensor(const std::vector<int>& lens,
+                           const std::vector<int>& strides,
+                           int offset,
+                           bool isPacked)
+    {
+        uint64_t max_value = miopen_type<T>{} == miopenHalf ? 5 : 17;
+
+        if(!isPacked)
+        {
+            std::vector<int> real_strides(strides.begin() + (strides.size() - lens.size()),
+                                          strides.end());
```

Comment:
`std::Vector<size_t>`, see #3393 for the explanation. There is absolutely no point to continue using `int`.